pacsman/pynetdicom_client.py

```diff
@@ -213,6 +265,42 @@ def fetch_images_as_files(self, series_id):
 
                 return series_path if os.path.exists(series_path) else None
 
+    def fetch_image_as_file(self, series_id, sop_instance_id):
+        series_path = os.path.join(self.dicom_dir, series_id)
+        with storage_scp(self.client_ae, series_path) as scp:
+            ae = AE(ae_title=self.client_ae,
+                    scu_sop_class=QueryRetrieveSOPClassList,
+                    transfer_syntax=[ExplicitVRLittleEndian])
+
+            extended_negotiation_info = []
+            for context in ae.presentation_contexts_scu:
+                negotiation = SCP_SCU_RoleSelectionNegotiation()
+                negotiation.sop_class_uid = context.abstract_syntax
+                negotiation.scu_role = False
+                negotiation.scp_role = True
+                extended_negotiation_info.append(negotiation)
+
+            with association(ae, self.pacs_url, self.pacs_port,
+                             ext_neg=extended_negotiation_info) as assoc:
+                dataset = Dataset()
+                dataset.SeriesInstanceUID = series_id
+                dataset.SOPInstanceUID = sop_instance_id
+                dataset.QueryRetrieveLevel = 'IMAGE'
+
+                if scp.is_alive():
+                    responses = assoc.send_c_move(dataset, scp.ae_title,
+                                                  query_model='S')
+                else:
+                    raise Exception(f'Storage SCP failed to start for series {series_id}')
+
+                for _ in checked_responses(responses):
+                    # just check response Status
+                    pass
+
+                filepath = scp.path_for_dataset_instance(dataset)
+                return filepath if os.path.exists(filepath) else None
+        return None
+
     def fetch_thumbnail(self, series_id):
```

Comment:
This should be renamed `fetch_middle_image_of_series` or something like that. I guess the `fetch_image_as_file` method is for when you want a specific image, but not the middle one?

Reply:
Makes sense, I see the distinction now.